docs/samples/Microsoft.ML.Samples/Dynamic/ValueMapping.cs

```diff
@@ -0,0 +1,281 @@
+            // 26  0-5yrs       Cat1
+            // 42  0-5yrs       Cat1
+            // 39  12+yrs       Cat3
+            // 34  0-5yrs       Cat1
+            // 35  6-11yrs      Cat2
+        }
+
+        ///<summary>
+        /// This example demonstrates the use of floating types as the key type for ValueMappingEstimator by mapping a float-to-string value.
+        /// The mapping looks like the following:
+        /// <list>
+        ///     <item>1.0 -> Cat1</item>
+        ///     <item>2.0 -> Cat2</item>
+        /// </list>
+        /// </summary>
+        public static void FloatToStringMappingExample(MLContext ml, IDataView trainData)
+        {
+            // Creating a list of keys based on the induced value from the dataset
+            // These lists are created by hand for the demonstration, but the ValueMappingEstimator does take an IEnumerable.
+            var inducedKeys = new List<float>()
+            {
+                1.0f, 
+                2.0f
+            };
+
+            // Sample list of associated string values
+            var inducedValues = new List<string>()
+            {
+                "Cat1",  
+                "Cat2"
+            };
+
+            var pipeline = new ValueMappingEstimator<float, string>(ml, inducedKeys, inducedValues, ("Induced", "InducedCategory"));
+
+            // The transformed data.
+            var transformedData = pipeline.Fit(trainData).Transform(trainData);
+
+            // Getting the data of the newly created column as an IEnumerable of SampleInfertDataWithFeatures.
+            var featuresColumn = ml.CreateEnumerable<SampleInfertDataWithInducedCategory>(transformedData, reuseRowObject: false);
+            
+            Console.WriteLine($"Example of mapping float->string");
+            Console.WriteLine($"Age\tInduced\tInducedCategory");
+            foreach (var featureRow in featuresColumn)
+            {
+                Console.WriteLine($"{featureRow.Age}\t{featureRow.Induced}\t{featureRow.InducedCategory}");
+            }
+
+            // Features column obtained post-transformation.
+            //
+            // Example of mapping float->string
+            // Age     Induced InducedCategory
+            // 26      1       Cat1
+            // 42      1       Cat1
+            // 39      2       Cat2
+            // 34      2       Cat2
+            // 35      1       Cat1
+        }
+
+        ///<summary>
+        /// This example demonstrates the use arrays as the values for the ValueMappingEstimator. It maps a set of keys that are type string
+        /// to a integer arrays of variable length.
+        /// The mapping looks like the following:
+        /// <list>
+        ///     <item>0-5yrs -> 1,2,3,4</item>
+        ///     <item>6-11yrs -> 5,6,7</item>
+        ///     <item>12+yrs -> 42, 32</item>
+        /// </list>
+        /// </summary>
+        public static void StringToArrayMappingExample(MLContext ml, IDataView trainData)
```

Comment:
It might make sense to put each example in its own file so it's easier to copy and paste.

Reply:
ok - I like that better, will update...

---
In reply to: [250459471](https://github.com/dotnet/machinelearning/pull/2222#discussion_r250459471) [](ancestors = 250459471)